### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Added Dockerfile and Docker configuration, refactored environment loading, and updated API and indexer entry points.

### What changed?

- Added a Dockerfile to build and run the API and indexer
- Refactored environment loading into a separate package
- Updated API and indexer entry points to use the new environment loading package
- Commented out database-related code in the GetBlocks handler

### How to test?

1. Build the Docker image:
   ```
   docker build -t indexer .
   ```
2. Run the container (add needed env vars in the command using `-e`):
   ```
   docker run -p 8080:8080 indexer
   ```
3. Verify that both the API and indexer services start without errors
4. Test the API endpoints to ensure they're functioning correctly

### CODE REVIEW

**Comment by iuwqyir** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/24?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/24?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/23?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/22?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: internal/env/load.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package env
+
+import (
+	"github.com/joho/godotenv"
+	"github.com/rs/zerolog/log"
```
Comment (AmineAfia) [inline internal/env/load.go:5]: not the logger from the previous stack?

**Review by AmineAfia — APPROVED** [review]

File: internal/env/load.go (hunk 1 above)
Comment (iuwqyir) [reply internal/env/load.go:5]: cannot be sure that env vars exist here if we're using dotenv

**Comment by iuwqyir** [comment]
### Merge activity

* **Sep 20, 6:05 AM EDT**: @iuwqyir started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/indexer/24).
